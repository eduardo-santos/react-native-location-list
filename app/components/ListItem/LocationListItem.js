import React from "react";
import { TouchableWithoutFeedback, View, Text } from "react-native";
import PropTypes from "prop-types";

import styles from "./styles";

const LocationListItem = ({ onPress, name, address }) => (
  <TouchableWithoutFeedback onPress={onPress}>
    <View style={styles.row}>
      <View style={styles.infoContainer}>
        <Text style={styles.title} numberOfLines={2}>
          {name}
        </Text>
        <Text style={styles.address} numberOfLines={2}>
          {address}
        </Text>
      </View>
      <View style={styles.infoContainerIndicator}>
        <Text style={styles.indicator}> + detalhes</Text>
      </View>
    </View>
  </TouchableWithoutFeedback>
);

LocationListItem.propTypes = {
  onPress: PropTypes.func,
  name: PropTypes.string,
  address: PropTypes.string
};

export default LocationListItem;
