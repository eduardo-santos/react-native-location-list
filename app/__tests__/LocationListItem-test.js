import React from "react";

import renderer from "react-test-renderer";
import LocationListItem from "../components/ListItem/LocationListItem";

it("renders correctly", () => {
  const tree = renderer.create(<LocationListItem />).toJSON();
  expect(tree).toMatchSnapshot();
});
