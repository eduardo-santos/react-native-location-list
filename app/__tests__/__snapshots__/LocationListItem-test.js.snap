// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "paddingHorizontal": 12,
      "paddingVertical": 12,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexWrap": "wrap",
        "justifyContent": "flex-start",
        "marginLeft": 12,
      }
    }
  >
    <Text
      numberOfLines={2}
      style={
        Object {
          "color": "#3C4146",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    />
    <Text
      numberOfLines={2}
    />
  </View>
  <View
    style={
      Object {
        "justifyContent": "flex-end",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#53B5E1",
          "fontSize": 16,
          "fontWeight": "normal",
        }
      }
    >
       + detalhes
    </Text>
  </View>
</View>
`;
